core/src/ancestor_hashes_service.rs

```diff
@@ -278,12 +312,14 @@ impl AncestorHashesService {
             );
         }
         time.stop();
-        if total_packets >= *max_packets {
-            if time.as_ms() > 1000 {
-                *max_packets = (*max_packets * 9) / 10;
-            } else {
-                *max_packets = (*max_packets * 10) / 9;
-            }
+        if total_packets >= packet_threshold.max_packets {
+            let threshold_update =
+                if time.as_ms() > DynamicPacketToProcessThreshold::TIME_THRESHOLD_MS {
```

Comment:
With `time` as an `Instant` and `TIME_THRESHOLD_MS` as a `Duration`, this becomes `if time.elapsed() > TIME_THRESHOLD_MS`, which you might consider encapsulating in a `DynamicPacketToProcessThreshold::should_update(...)` or better yet, stick this whole block in an `...::update_if(...)`

Reply:
Fixed.